### PR TITLE
test: add tests for tenant service

### PR DESCRIPTION
**Pull request by affo**

### CODE REVIEW

File: testing/tenant.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		{
+			name: "CascadingDelete",
+			fn:   CascadingDelete,
+		},
+		// TODO(affo): add tests.
+		//  - cascading delete urm: for every resource, if you delete it you delete the urm.
+		//  - cascading delete user: if you delete a user you delete every associated resource.
+		//    This should happen only  when that  user is the only member/owner of that resource.
+		//  - cascading delete organization: if you delete an org all its content gets dropped.
+		//    NOTE: do we need a tombstone and an expiry period?
```
Comment (affo) [inline testing/tenant.go]: some tests ideas

File: kv/tenant_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			t.Fatalf("error populating dashboards: %v", err)
+		}
+	}
+	for _, urm := range f.UserResourceMappings {
+		if err := svc.CreateUserResourceMapping(ctx, urm); err != nil {
+			t.Fatalf("error populating urms: %v", err)
+		}
+	}
+
+	return tenantService{
```
Comment (lyondhill) [inline kv/tenant_test.go]: You could probably just return svc here if you wanted.

**Comment by affo** [comment]
@lyondhill I think this is now ready to go 👍

File: kv/tenant_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	itesting "github.com/influxdata/influxdb/testing"
+	"go.uber.org/zap/zaptest"
+)
+
+func TestBoltTenantService(t *testing.T) {
+	t.Skip("some tests fail using `*kv.Service` as `influxdb.TenantService`, see the NOTEs in `testing/tenant.go`.")
+	itesting.TenantService(t, initBoltTenantService)
+}
+
+func TestBoltNewTenantService(t *testing.T) {
```
Comment (lyondhill) [inline kv/tenant_test.go]: this function should be in the tenant/tenant_test.go

Comment (affo) [reply kv/tenant_test.go]: yes, you are right 😓

**Pull request merged**